core/templates/domain/collection/collection.model.ts

```diff
@@ -32,35 +32,40 @@ interface ExplorationIdToNodeIndexMap {
 }
 
 export interface CollectionBackendDict {
-  'id': string;
-  'title': string;
-  'objective': string;
-  'language_code': string;
-  'tags': string[];
-  'schema_version': number;
+  // On creating a new Collection, properties below are always
```

Comment:
I think there is a need to explain exactly why these properties need to be initialized to `null` and what it implies conceptually.

Reply:
Updated PTAL!

Reply:
Ummm, take [this](https://github.com/oppia/oppia/pull/12962/files#diff-9300adbb31e5176355bc7cf7c61f1d4caabe352395a4f9eaffb20f891bd7d007) explanation forexample, we could have said that those properties are initialized to `null` but that explanation is not sufficient as Sean mentioned. Does that make sense?

Reply:
@EeshaArif Creating a collection is just like creating a exploration.A non-logged in user cannot create a collection.I don't think we should mention something like `the properties are null if the user is not logged in`.They would still be null till the moment user clicks `Create Collection` button.

Reply:
No, I think you may have misunderstood. I wasn't implying to mention the same comment, was giving an example oh how just stating that the properties are initialized to `null` is not a sufficient explanation. If you take a look at the PR mentioned above. There are some comments on a removed file that may give you some idea on what sort of explanation may be considered sufficient.

Reply:
Done, please take a look